### ISSUE TITLE
Death Stranding Director's Cut patcher not working under wine

### ISSUE THREAD

**Issue by jibsaramnim** [RoseTheFlower/UltrawideIndex#25]
Hello! First off, thank you so much for everything that you've done to make these games more accessible for those of us who have ultrawide displays. I truly appreciate it! Onto my question;

---

With the original version of Death Stranding I was able to use your patcher even though I play the game under Linux. I would launch the game through Steam, then separately launch the patcher within the same wine prefix, and it would work as expected.

However, with the Director's Cut edition I'm somehow not able to get it to work. Do you happen to know if something changed in the way you have implemented the hook/listener for the function keys? 

It would be really nice if there was a way to get it to work again. I purchased the Director's Cut version mostly to support the developer and I was hopeful it would just come with 32:9 support out of the box, but that ended up not being the case, and unfortunately I have thus far not been able to get your patcher to work again too, so I've not been able to play again.

Anyway, I know you officially support only Windows, but for those of us who do not use this and do have an ultrawide, it would be really nice if whatever changed between the original and DC versions could be adjusted so that us Linux folk could play the game in all its ultrawide goodness again. Either way, thanks so much!

**Comment by RoseTheFlower** [comment]
Hi. Could you confirm that the original trainer still works as of today and the version of Wine or Proton that you're using?

Is [this suggestion](https://community.pcgamingwiki.com/files/file/1458-red-dead-redemption-2-ultrawide-multimonitor-fov-mod/?do=findComment&comment=2281&tab=comments) of any use?

**Comment by jibsaramnim** [comment]
Sorry for the slightly delayed response,  I had to reinstall the non-DC version. I just tried again and, somehow, I can't get the patcher to work for the non-DC version either. 

I am indeed launching the patcher in the same way as the link you have provided, and checking with `taskmgr` both the patcher and game are seen in the processes list, so I'm not sure what might be going on. I used to play this way, though admittedly it has been a while (>6 months at least).

**Comment by RoseTheFlower** [comment]
The F3 and F4 functions aren't conditional, so they will get highlighted even if you run the trainer by itself.

I sent a message to the person who made the linked comment to inquire if they've run any of my more recent trainers.

**Comment by jibsaramnim** [comment]
Thank you kindly! I'm not sure what may have changed -- it did work in the past for me (for the non-DC version), even though that too seems to not work anymore now. Thank you so much for looking into it, I really appreciate it!

**Comment by RoseTheFlower** [comment]
Could you try following [this guide](https://old.reddit.com/r/SteamDeck/comments/wklnqx/a_guide_to_set_up_trainerscheats_through_steam/)? If not on the Deck, perhaps there is something similar for the other distros.

**Comment by rialpa84** [comment]
I just thought I'd let you know that this does work on Linux, under Wine, as long as it and the game are ran from the same prefix. I used the "Run EXE inside of Wine prefix" feature included in Lutris. The guide you posed should work, too, in a more automated manner.
![Screenshot_32](https://github.com/RoseTheFlower/UltrawideIndex/assets/84974031/8dfc8430-090e-4e12-8de6-f04bcad11575)

**Comment by RoseTheFlower** [comment]
Thanks for the report! The job of digging into so many games and maintaining over 100 mods is quite tough. If I were to also test them on Mac and Linux, or on each specific Linux distro, it would literally multiply the amount of work I have to do and distract from getting on new titles. I'm glad there are users that share their knowledge to make up this.

**Comment by rialpa84** [comment]
And thank you for the tool. This is one of those little annoyances that tends trip up people the first time they come across it and why things like Lutris and SteamTinkerLauncher make lives easier. The alternative is launching the tool with something like this:
`
WINEPREFIX="/external_disk/wine/egs" "/home/$USERNAME/.local/share/Steam/compatibilitytools.d/GE-Proton8-9/files/bin/wine" "/external_disk/games/pc/windows/Epic Games/DeathStrandingDC/Death Stranding Director's Cut - Display Mods v1.0 - Epic.exe"`

That command specifically sets the WINEPREFIX variable to use my generic Epic Games prefix and uses the same Wine binaries (GE-Proton8-9 in this instance) that the game will be using. 

If you just use `wine /path/to/Death Stranding Director's Cut - Display Mods v1.0 - Epic.exe` you'll use the system installed Wine which probably isn't the one with DXVK, vkd3d-proton, and all the other goodies that make games run well on Linux.

**Comment by Calinou** [comment]
For context, I've tried to run the Jedi Fallen Order ultrawide EXE in the same Proton Experimental prefix as the game itself (running via Heroic), but since the "Run EXE inside prefix" option does nothing for me[^1], I tried to resort to the command line:

```bash
STEAM_COMPAT_DATA_PATH="/home/hugo/.local/share/heroic/prefixes/Jedi Fallen Order" STEAM_COMPAT_CLIENT_INSTALL_PATH="/home/hugo/.local/share/Steam" "/home/hugo/.local/share/Steam/steamapps/common/Proton - Experimental/proton" Fallen\ Order\ Ultrawide\ 2.1.exe
```

Unfortunately, the trainer still can't "see" the running game when I do this (I launched the game first).

[^1]: Literally nothing happens, even though I'm using the AppImage and [not the Flatpak which has known issues with this](https://github.com/Heroic-Games-Launcher/HeroicGamesLauncher/issues/2766).

**Comment by GrabbenD** [comment]
**Heads-up for future readers**

* If you're using [WINE's new Wayland driver](https://gitlab.winehq.org/wine/wine/-/releases/wine-9.0#wayland-driver) you _might_ encounter problems with older /or/ some games:
https://bugs.winehq.org/show_bug.cgi?id=56258
https://bugs.winehq.org/show_bug.cgi?id=56172
https://bugs.winehq.org/show_bug.cgi?id=56273

* I personally had no issues with `WINEDLLOVERRIDES` in a lot of new games like High On Life.
   However, if you're still having issues, you might find this helpful (I only had to do this for Dead Island 2):
   
   > https://github.com/Lyall/DeadIsland2Fix/issues/4
   > 
   > 1. Follow the installation instructions for `DeadIsland2Fix_v_1_0_2.zip` (meaning `DeadIsland2Fix.asi` + `DeadIsland2Fix.ini` + `winmm.dll` should be placed in: `<GameDrive>/DeadIsland/Binaries/Win64`)
   > 
   > 2. Create `<GameDrive>/DeadIsland/Binaries/Win64/winmm.ini` with:
   >     ```ini
   >     [GlobalSets]
   >     DontLoadFromDllMain=0
   >     ```
   > 
   > 3. Use `WINEDLLOVERRIDES="winmm.dll=n,b" %command%` to start the (steam) game.
   >     Alternatively, with Bottles it's as simple as adding `winmm` as `Native,then Builtin` in `Bottle > Settings > DLL Overrides`:
   >     ![Bottle](https://github.com/Lyall/DeadIsland2Fix/assets/45666572/4c66e5c1-d91c-4bd5-b20d-73955cb3e72b)
   > 
   > 4. Lastly, I launch the game using: `<GameDrive>/DeadIsland/Binaries/Win64/DeadIsland-Win64-Shipping.exe`
   > 
   > ---
   > 
   > For future readers there's also another (potential) fix; projects that uses ThirteenAG's codebase (Ultimate ASI Loader), it's possible to simply rename the `winmm.ini` to one of supported filenames from the README like `dsound.dll` (but with Dead Island 2 this particular filename caused conflicts hence the crashing): 
   > https://github.com/ThirteenAG/Ultimate-ASI-Loader

**Comment by RoseTheFlower** [comment]
>Use WINEDLLOVERRIDES="winmm.dll=n,b" %command% to start the (steam) game.

I don't think this is relevant to the Death Stranding fix, as it's a trainer, not a DLL to inject.

**Comment by GrabbenD** [comment]
@RoseTheFlower I've posted these workarounds/pointers [^](https://github.com/RoseTheFlower/UltrawideIndex/issues/25#issuecomment-1931780893) here since you've linked to this Github Issue in the instructions page (and knowing all of that would have tremendously helped me in the beginning):
https://github.com/RoseTheFlower/UltrawideIndex/blob/main/instructions/linux.md

**Comment by RoseTheFlower** [comment]
>since you've linked to this Github Issue in the instructions page 

That's in the sentence on targeting a running game process, which is what this trainer does. A DLL placed in the game folder injects in a different way and I have a separate section for that. If you believe that any of the information is inaccurate or it could be improved, you can create a pull request and we would discuss there. Given the number of Linux distros and various configurations, the instructions have to be quite broad though.
